pkg/lambda/options.go

```diff
@@ -0,0 +1,105 @@
+package lambda
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"go.uber.org/multierr"
+
+	"github.com/atlassian/gostatsd/internal/awslambda/extension/api"
+	"github.com/atlassian/gostatsd/pkg/statsd"
+)
+
+const (
+	ParamRuntimeAPI        = `lambda-runtime-api`
+	ParamEntryPointName    = `lamda-entrypoint-name`
+	ParamEnableManualFlush = `enable-manual-flush`
+	ParamTelemetryAddr     = `telemetry-addr`
+)
+
+// Options is used to provide overrides when calling `NewExtension`
+type Options struct {
+	// Server is extended by the extension
+	// This value is required, no defaults are provided.
+	Server *statsd.Server
+	// Logger is used within the extension.
+	// This value is required, no defaults are provided.
+	Logger logrus.FieldLogger
+	// RuntimeAPI is AWS service that is responsible for orchestrating
+	// the extension with the other Lambda's that are part of the deployment.
+	// This will set to `env:AWS_LAMBDA_RUNTIME_API` by default as per the docs,
+	// but can be overriden for testing purposes and validation.
+	RuntimeAPI string
+	// ExecutableName is the full file name of the lambda extension that is
+	// used to validate the bootstrap process.
+	ExecutableName string
+	// EnableManualFlush allows for Lambda's
+	// to control when the forwarder would send the accumlated metrics.
+	EnableManualFlush bool
+	// TelemetryAddr is used by the AWS runtime to publish
+	// events back to service.
+	TelemetryAddr string
+}
+
+func NewDefaultOptions() Options {
```

Comment:
Can we call this `NewOptionsFromEnvironment`?  It makes it more explicit this isn't purely defaults, and that it's not a pure function generally.